projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js

```diff
@@ -4,222 +4,263 @@ import apiFetch from '@wordpress/api-fetch';
+						'jetpack'
+					)
+				);
+				setLoadingCompletion( false );
+			} );
+	};
+
+	// Waiting state means there is nothing to be done until it resolves
+	const waitingState = completionFinished || loadingCompletion || loadingCategories;
+	const nbCharactersNeeded = numberOfCharactersNeeded - content.length;
+	if ( ! waitingState ) {
+		if ( containsAiUntriggeredParapgraph( contentBefore ) ) {
+			if ( ! errorMessage ) {
+				setErrorMessage(
+					/** translators: This will be an error message when multiple Open AI paragraph blocks are triggered on the same page. */
+					__( 'Waiting for the previous AI paragraph block to finish', 'jetpack' )
+				);
+			}
+		} else if (
+			content.length < numberOfCharactersNeeded &&
+			needsMoreCharacters !== nbCharactersNeeded
+		) {
+			setErrorMessage(
+				sprintf(
+					/** translators: First placeholder is a number of more characters we need */
+					__(
+						'Please write a longer title or a few more words in the opening preceding the AI block. Our AI model needs %1$d more characters.',
+						'jetpack'
+					),
+					nbCharactersNeeded
+				)
 			);
+			setNeedsMoreCharacters( nbCharactersNeeded );
+		} else if ( needsMoreCharacters !== 0 && content.length >= numberOfCharactersNeeded ) {
+			setErrorMessage(
+				/** translators: This is to retry to complete the text */
+				__( 'Ready to retry', 'jetpack' )
+			);
+			setShowRetry( true );
+			setNeedsMoreCharacters( 0 );
+		} else if ( needsMoreCharacters === 0 && ! showRetry ) {
+			getSuggestionFromOpenAI();
 		}
-	}, [ setAttributes, attributes ] ); // eslint-disable-line react-hooks/exhaustive-deps
+	}
 
 	return (
 		<div { ...useBlockProps() }>
-			{ ! loadingCompletion && errorMessage && (
+			{ ! loadingCompletion && ! loadingCategories && errorMessage && (
 				<Placeholder label={ __( 'AI Paragraph', 'jetpack' ) } instructions={ errorMessage }>
-					<Button
-						variant="primary"
-						onClick={ () => {
-							getSuggestionFromOpenAI(
-								setAttributes,
-								formattedPrompt,
-								setLoadingCompletion,
-								setErrorMessage
-							);
-						} }
-					>
-						{ __( 'Retry', 'jetpack' ) }
-					</Button>
+					{ showRetry && (
+						<Button variant="primary" onClick={ () => getSuggestionFromOpenAI() }>
+							{ __( 'Retry', 'jetpack' ) }
+						</Button>
+					) }
 				</Placeholder>
 			) }
-			{ attributes.content && ! loadingCompletion && (
-				<div>
-					<div className="content">
-						<RawHTML>{ attributes.content.trim().replaceAll( '\n', '<br/>' ) }</RawHTML>
-					</div>
-				</div>
+
+			{ attributes.content && (
+				<ShowLittleByLittle
+					showAnimation={ showAnimation }
+					html={ attributes.content.trim().replaceAll( '\n', '<br/>' ) }
+				/>
 			) }
-			{ loadingCompletion && (
+
+			{ ! attributes.content && ( loadingCompletion || loadingCategories ) && (
 				<div style={ { padding: '10px', textAlign: 'center' } }>
 					<Spinner
```

Comment:
I didn't know about `Spinner`!  I implemented my own with my modal PR 🤦‍♂️

Reply:
That was already there, I would not have known about it othgerwise... 
https://developer.wordpress.org/block-editor/reference-guides/components/spinner/